### DISCUSSION TITLE
PAR field from gotm/light upside down

### DISCUSSION THREAD

**Discussion by janko-om** [fabm-model/fabm#97] [Q&A]
Hi!

I use gotm/light to calculate the PAR field for my ecosystem model (nersc/ecosmo). Unfortunately first tests show phytoplankton growth at the seafloor and no growth at the surface, leading to the conclusion that the PAR field must be supplied to ECOSMO upside down. The host is SCHISM v5.11 (with _FABM_VERTICAL_BOTTOM_TO_SURFACE_) and I use FABMv2.1.3. The PAR field written to the output NetCDF file strangely has the correct vertical orientation.

Any suggestions where to start looking for the cause?

**Comment by jornbr** [comment]
Hi, that's is certainly surprising. Does this occur with the latest, unmodified FABM code? Specifically, is `src/drivers/schism/fabm_driver.h` identical to https://github.com/fabm-model/fabm/blob/master/src/drivers/schism/fabm_driver.h?

If yes, is this at all familiar to other SCHISM users? Have there been previous SCHISM-FABM simulations with the same ECOSMO/light models where this issue did not occur? And if so, is this associated with specific SCHISM or FABM versions, for instance, did it occur only after updating one or the other?

**Reply by janko-om** [reply]
Yes, this occurs with the latest FABM code. I however use a ecosmo.F90 that currently isn't included in the FABM code. The **`src/drivers/schism/fabm_driver.h`** is identical to the current master branch.

I previously used SCHISMv5.9 with FABMv0, but did not invoke gotm/light because the light conditions where calculated by an internal light model using get_light and get_light_extinction. The differences now are:

- SCHISM code is updated to v5.11
- fabm_schism.F90 is slightly modified because of some work in progress, but this isn't anything major I think: [fabm_schism.F90](https://github.com/user-attachments/files/17699955/fabm_schism_JK.txt)
- FABMv2.1.3 cloned in September 2024, with no modification of the core code
- gotm/light is now used to provide PAR to ECOSMO
- I adapted ecosmo.F90 for use with FABMv2 and implemented light extinction using `self%add_to_aggregate_variable(standard_variables%attenuation_coefficient_of_photosynthetic_radiative_flux)`

I'm currently the only user of SCHISM-FABM with ECOSMO+gotm/light. But @platipodium mentioned they were also having problems with the light field?

**Comment by jornbr** [comment]
Just wondering - was this issue ultimately resolved, e.g., with support from @platipodium? Or is this something that needs to investigated on the FABM side?

**Reply by janko-om** [reply]
Hi! Sorry, I'm currently not working on this and forgot to close the question. The problem persists, but the orientation of the PAR field was excluded as the source of the problem. No further work from the FABM side necessary, thanks!
